### DISCUSSION TITLE
dcim.napalm_read_device permission

### DISCUSSION THREAD

**Discussion by PieterL75** [netbox-community/netbox#6259] [Q&A] [answered]
Hey,

I'm trying out the Napalm integration. It works perfect for my 'all rights' user.
But when I use a user that has limited rights, he is not able to see any 'Napalm' data.

The guide says that my user needs the 'dcim.napalm_read_device' permission.
But I'm not able to find the the 'Home › Users › Permissions › Add permission' the 'Napalm Read Device' permission.

Can you guide me on how to set that ?

Pieter

**Comment by jeremystretch** [comment, marked as answer]
You need to create a permission on the Device model with the nonstandard `napalm_read` action. This is mentioned in [the permissions documentation](https://netbox.readthedocs.io/en/stable/administration/permissions/#actions).

![Screenshot_2021-04-23 Add permission NetBox](https://user-images.githubusercontent.com/13487278/115887104-9b37d100-a41f-11eb-801f-1134bdd90b98.png)

**Reply by PieterL75** [reply]
Jeremy,

I was only looking in the https://netbox.readthedocs.io/en/stable/additional-features/napalm/ documentation..
Thank you for pointing that out !

I added the additional action, and it all works fine now

Pieter
